<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integration between Multiple spatial datasets and single cell sequencing data â€¢ Giotto Suite</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Integration between Multiple spatial datasets and single cell sequencing data">
<meta property="og:description" content="Giotto">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-BCXT5392QC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BCXT5392QC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">Giotto Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0.991</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Start
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/getting_started.html#installation">Installation</a>
    </li>
    <li>
      <a href="../../articles/getting_started.html#getting-started">Getting Started</a>
    </li>
    <li>
      <a href="../../articles/getting_started.html#analyses">Analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/datasets_overview.html#array-based-datasets-1">Array based</a>
    </li>
    <li>
      <a href="../../articles/datasets_overview.html#single-cell-1">Single-cell</a>
    </li>
    <li>
      <a href="../../articles/datasets_overview.html#subcellular-datasets-1">Subcellular</a>
    </li>
    <li>
      <a href="../../articles/datasets_overview.html#protein-multiplexing-1">Protein multiplexing</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../articles/contribute.html">Contribute</a>
</li>
<li>
  <a href="../../articles/giotto_news.html">News</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/faqs.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../../articles/github_issues.html">GitHub Issues Guidelines</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Try Giotto
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Docker</li>
    <li class="dropdown-header">Binder</li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RubD/Giotto/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Visium_prostate_integration_Sep30_21_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integration between Multiple spatial datasets and single cell sequencing data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RubD/Giotto/blob/master/vignettes/vignettes/Visium_prostate_integration_Sep30_21.Rmd"><code>vignettes/vignettes/Visium_prostate_integration_Sep30_21.Rmd</code></a></small>
      <div class="hidden name"><code>Visium_prostate_integration_Sep30_21.Rmd</code></div>

    </div>

    
    
<div id="set-up-giotto-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up-giotto-environment" class="anchor"></a>Set up Giotto Environment</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/">Giotto</a></span><span class="op">)</span>

<span class="co"># 1. set working directory</span>
<span class="va">results_folder</span> <span class="op">=</span> <span class="st">'path/to/result'</span>

<span class="co"># 2. set giotto python path</span>
<span class="co"># set python path to your preferred python version path</span>
<span class="co"># set python path to conda env/bin/ directory if manually installed Giotto python dependencies by conda</span>
<span class="co"># python_path = '/path_to_conda/.conda/envs/giotto/bin/python'</span>
<span class="co"># set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment</span>
<span class="va">python_path</span> <span class="op">=</span> <span class="va">null</span>
<span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">python_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../../reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># 3. create giotto instructions</span>
<span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="../../reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>,
                                  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,
                                  show_plot <span class="op">=</span> <span class="cn">FALSE</span>,
                                  python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></code></pre></div>
</div>
<div id="dataset-explanation" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset-explanation" class="anchor"></a>Dataset explanation</h2>
<p><a href="https://www.10xgenomics.com/spatial-transcriptomics/">10X genomics</a> recently launched a new platform to obtain spatial expression data using a Visium Spatial Gene Expression slide.</p>
<p>The Visium Cancer Prostate data to run this tutorial can be found <a href="https://www.10xgenomics.com/resources/datasets/human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe-1-standard-1-3-0">here</a> The Visium Normal Prostate data to run this tutorial can be found <a href="https://www.10xgenomics.com/resources/datasets/normal-human-prostate-ffpe-1-standard-1-3-0">here</a></p>
<p>Visium technology:<br><img src="../inst/images/general_figs/visium_technology.png" style="width:50.0%"></p>
<p>High resolution png from original tissue:<br><img src="../inst/images/general_figs/Visium_FFPE_Human_Normal_Prostate_image.png" style="width:50.0%"><img src="../inst/images/general_figs/Visium_FFPE_Human_Prostate_Cancer_image.png" style="width:50.0%"></p>
</div>
<div id="part-1-create-giotto-objects-and-join" class="section level2">
<h2 class="hasAnchor">
<a href="#part-1-create-giotto-objects-and-join" class="anchor"></a>Part 1: Create Giotto objects and join</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dataDir</span> <span class="op">&lt;-</span> <span class="st">'path/to/data'</span>
<span class="co">## obese upper</span>
<span class="va">N_pros</span> <span class="op">=</span> <span class="fu"><a href="../../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span>
    visium_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">dataDir</span>,<span class="st">'/Visium_FFPE_Human_Normal_Prostate/'</span><span class="op">)</span>,
    expr_data <span class="op">=</span> <span class="st">'raw'</span>,
    png_name <span class="op">=</span> <span class="st">'tissue_lowres_image.png'</span>,
    gene_column_index <span class="op">=</span> <span class="fl">2</span>,
    instructions <span class="op">=</span> <span class="va">instrs</span>
<span class="op">)</span>

<span class="co">## obese lower</span>
<span class="va">C_pros</span> <span class="op">=</span> <span class="fu"><a href="../../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span>
    visium_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">dataDir</span>,<span class="st">'/Visium_FFPE_Human_Prostate_Cancer/'</span><span class="op">)</span>,
    expr_data <span class="op">=</span> <span class="st">'raw'</span>,
    png_name <span class="op">=</span> <span class="st">'tissue_lowres_image.png'</span>,
    gene_column_index <span class="op">=</span> <span class="fl">2</span>,
    instructions <span class="op">=</span> <span class="va">instrs</span>
<span class="op">)</span>
<span class="co"># join giotto objects</span>
<span class="co"># joining with x_shift has the advantage that you can join both 2D and 3D data</span>
<span class="co"># x_padding determines how much distance is between each dataset</span>
<span class="co"># if x_shift = NULL, then the total shift will be guessed from the giotto image</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/joinGiottoObjects.html">joinGiottoObjects</a></span><span class="op">(</span>gobject_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">N_pros</span>, <span class="va">C_pros</span><span class="op">)</span>,
    gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'NP'</span>, <span class="st">'CP'</span><span class="op">)</span>,
    join_method <span class="op">=</span> <span class="st">'x_shift'</span>, x_padding <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>


<span class="co"># join info is stored in this slot</span>
<span class="co"># simple list for now</span>
<span class="va">testcombo</span><span class="op">@</span><span class="va">join_info</span>

<span class="co">#fDataDT(testcombo)</span>
<span class="co">#pDataDT(testcombo)</span>
<span class="co">#showGiottoImageNames(testcombo)</span>
<span class="co">#showGiottoSpatLocs(testcombo)</span>
<span class="co">#showGiottoExpression(testcombo)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this plots all the images by list_ID</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, cell_color <span class="op">=</span> <span class="st">'in_tissue'</span>,
    show_image <span class="op">=</span> <span class="cn">T</span>, image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NP-image"</span>, <span class="st">"CP-image"</span><span class="op">)</span>,
    group_by <span class="op">=</span> <span class="st">'list_ID'</span>, point_alpha <span class="op">=</span> <span class="fl">0.5</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"1a_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/1a_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this plots one selected image</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, cell_color <span class="op">=</span> <span class="st">'in_tissue'</span>,
    show_image <span class="op">=</span> <span class="cn">T</span>, image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NP-image"</span><span class="op">)</span>, point_alpha <span class="op">=</span> <span class="fl">0.3</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"1b_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/1b_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this plots two selected images</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, cell_color <span class="op">=</span> <span class="st">'in_tissue'</span>,
    show_image <span class="op">=</span> <span class="cn">T</span>, image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"NP-image"</span>, <span class="st">"CP-image"</span><span class="op">)</span>,
    point_alpha <span class="op">=</span> <span class="fl">0.3</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"1c_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/1c_plot.png" style="width:50.0%"></p>
</div>
<div id="part-2-process-giotto-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#part-2-process-giotto-objects" class="anchor"></a>Part 2: Process Giotto Objects</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset on in-tissue spots</span>
<span class="va">metadata</span> <span class="op">=</span> <span class="fu"><a href="../../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span>
<span class="va">in_tissue_barcodes</span> <span class="op">=</span> <span class="va">metadata</span><span class="op">[</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/subsetGiotto.html">subsetGiotto</a></span><span class="op">(</span><span class="va">testcombo</span>, cell_ids <span class="op">=</span> <span class="va">in_tissue_barcodes</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## filter</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    expression_threshold <span class="op">=</span> <span class="fl">1</span>,
    feat_det_in_min_cells <span class="op">=</span> <span class="fl">50</span>,
    min_det_feats_per_cell <span class="op">=</span> <span class="fl">500</span>,
    expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'raw'</span><span class="op">)</span>,
    verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>

<span class="co">## normalize</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, scalefactor <span class="op">=</span> <span class="fl">6000</span><span class="op">)</span>
<span class="co">#testcombo &lt;- normalizeGiotto(gobject = testcombo, norm_methods = 'pearson_resid')</span>

<span class="co">## add gene &amp; cell statistics</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, expression_values <span class="op">=</span> <span class="st">'raw'</span><span class="op">)</span>

<span class="va">fmeta</span> <span class="op">=</span> <span class="fu"><a href="../../reference/fDataDT.html">fDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span>
<span class="va">testfeats</span> <span class="op">=</span> <span class="va">fmeta</span><span class="op">[</span><span class="va">perc_cells</span> <span class="op">&gt;</span> <span class="fl">20</span> <span class="op">&amp;</span> <span class="va">perc_cells</span> <span class="op">&lt;</span> <span class="fl">50</span><span class="op">]</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">110</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span>

<span class="fu"><a href="../../reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">testcombo</span>, feats <span class="op">=</span> <span class="va">testfeats</span>, cluster_column <span class="op">=</span> <span class="st">'list_ID'</span>, save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"2a_plot"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">testcombo</span>, selected_feats <span class="op">=</span> <span class="va">testfeats</span>, metadata_cols <span class="op">=</span> <span class="st">'list_ID'</span>, save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"2b_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/2a_plot.png" style="width:50.0%"><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/2b_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## visualize</span>
<span class="co">#fDataDT(testcombo)</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, group_by <span class="op">=</span> <span class="st">'list_ID'</span>, cell_color <span class="op">=</span> <span class="st">'nr_feats'</span>, color_as_factor <span class="op">=</span> <span class="cn">F</span>, point_size <span class="op">=</span> <span class="fl">0.75</span>, save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"2c_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/2c_plot.png" style="width:50.0%"></p>
</div>
<div id="part-3-dimention-reduction" class="section level2">
<h2 class="hasAnchor">
<a href="#part-3-dimention-reduction" class="anchor"></a>Part 3: Dimention Reduction</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## PCA ##</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span><span class="op">)</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale_unit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="../../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">testcombo</span>, ncp <span class="op">=</span> <span class="fl">30</span>, save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"3a_screeplot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/3a_screeplot.png" style="width:50.0%"></p>
</div>
<div id="part-4-clustering" class="section level2">
<h2 class="hasAnchor">
<a href="#part-4-clustering" class="anchor"></a>Part 4: Clustering</h2>
<div id="without-integration" class="section level3">
<h3 class="hasAnchor">
<a href="#without-integration" class="anchor"></a>4.1 Without Integration</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## WITHOUT INTEGRATION ##</span>
<span class="co"># --------------------- #</span>

<span class="co">## cluster and run UMAP ##</span>
<span class="co"># sNN network (default)</span>
<span class="fu"><a href="../../reference/showGiottoDimRed.html">showGiottoDimRed</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    dim_reduction_to_use <span class="op">=</span> <span class="st">'pca'</span>, dim_reduction_name <span class="op">=</span> <span class="st">'pca'</span>,
    dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>

<span class="co"># Leiden clustering</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, resolution <span class="op">=</span> <span class="fl">0.2</span>, n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>

<span class="co"># UMAP</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span>

<span class="fu"><a href="../../reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, show_NN_network <span class="op">=</span> <span class="cn">T</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.1a_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.1a_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, group_by <span class="op">=</span> <span class="st">'list_ID'</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,
    point_size <span class="op">=</span> <span class="fl">1.5</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.1b_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.1b_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/spatDimPlot2D.html">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.1c_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.1c_plot.png" style="width:50.0%"></p>
</div>
<div id="with-harmony-integration" class="section level3">
<h3 class="hasAnchor">
<a href="#with-harmony-integration" class="anchor"></a>4.2 With Harmony integration</h3>
<p>Harmony is a integration algorithm developed by <a href="https://www.nature.com/articles/s41592-019-0619-0">Korsunsky, I. et al.</a>. It was designed for integration of single cell data but also work well on spatial datasets.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## WITH INTEGRATION ##</span>
<span class="co"># --------------------- #</span>

<span class="co">## data integration, cluster and run UMAP ##</span>

<span class="co"># harmony</span>
<span class="co">#library(devtools)</span>
<span class="co">#install_github("immunogenomics/harmony")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">harmony</span><span class="op">)</span>

<span class="co">#pDataDT(testcombo)</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runGiottoHarmony.html">runGiottoHarmony</a></span><span class="op">(</span><span class="va">testcombo</span>, vars_use <span class="op">=</span> <span class="st">'list_ID'</span>, do_pca <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>


<span class="co">## sNN network (default)</span>
<span class="co">#showGiottoDimRed(testcombo)</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    dim_reduction_to_use <span class="op">=</span> <span class="st">'harmony'</span>, dim_reduction_name <span class="op">=</span> <span class="st">'harmony'</span>, name <span class="op">=</span> <span class="st">'NN.harmony'</span>,
    dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>

<span class="co">## Leiden clustering</span>
<span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    network_name <span class="op">=</span> <span class="st">'NN.harmony'</span>, resolution <span class="op">=</span> <span class="fl">0.2</span>, n_iterations <span class="op">=</span> <span class="fl">1000</span>, name <span class="op">=</span> <span class="st">'leiden_harmony'</span><span class="op">)</span>

<span class="co"># UMAP dimension reduction</span>
<span class="co">#showGiottoDimRed(testcombo)</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">testcombo</span>, dim_reduction_name <span class="op">=</span> <span class="st">'harmony'</span>, dim_reduction_to_use <span class="op">=</span> <span class="st">'harmony'</span>, name <span class="op">=</span> <span class="st">'umap_harmony'</span><span class="op">)</span>
<span class="fu"><a href="../../reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    dim_reduction_name <span class="op">=</span> <span class="st">'umap_harmony'</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_harmony'</span>, show_NN_network <span class="op">=</span> <span class="cn">T</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.2a_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.2a_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, group_by <span class="op">=</span> <span class="st">'list_ID'</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_harmony'</span>,
    point_size <span class="op">=</span> <span class="fl">1.5</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.2b_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.2b_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/spatDimPlot2D.html">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    dim_reduction_to_use <span class="op">=</span> <span class="st">'umap'</span>, dim_reduction_name <span class="op">=</span> <span class="st">'umap_harmony'</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_harmony'</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4.2c_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4.2c_plot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># compare to previous results</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, 
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_w_o_integration_plot"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
    cell_color <span class="op">=</span> <span class="st">'leiden_harmony'</span>,
    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_w_integration_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4_w_o_integration_plot.png" style="width:50.0%"><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/4_w_integration_plot.png" style="width:50.0%"></p>
</div>
</div>
<div id="part-5-cell-type-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#part-5-cell-type-annotation" class="anchor"></a>Part 5: Cell type annotation</h2>
<p>Visium spatial transcriptomics does not provide single-cell resolution, making cell type annotation a harder problem. Giotto provides several ways to calculate enrichment of specific cell-type signature gene list:<br>
- PAGE<br>
- hypergeometric test - Rank - <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02362-7">DWLS Deconvolution</a></p>
<p>This is also the easiest way to integrate Visium datasets with single cell data. Example shown here is from <a href="https://pubmed.ncbi.nlm.nih.gov/33032611/">Ma et al.</a> from two prostate cancer patients. The raw dataset can be found <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157703">here</a> Giotto_SC is processed variable in the <a href="https://giottosuite.com/articles/singlecell_prostate_standard_Sep29_21.html">single cell RNAseq tutorial</a></p>
<div id="page-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#page-enrichment" class="anchor"></a>5.1 PAGE enrichment</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create PAGE matrix</span>
<span class="co"># PAGE matrix should be a binary matrix with each row represent a gene marker and each column represent a cell type</span>
<span class="co"># markers_scran is generated from single cell analysis ()</span>
<span class="va">markers_scran</span> <span class="op">=</span> <span class="fu"><a href="../../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject<span class="op">=</span><span class="va">giotto_SC</span>, method<span class="op">=</span><span class="st">"scran"</span>,
                                       expression_values<span class="op">=</span><span class="st">"normalized"</span>, cluster_column<span class="op">=</span><span class="st">'prostate_labels'</span>, min_feats<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">top_markers</span> <span class="op">&lt;-</span> <span class="va">markers_scran</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">10</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"cluster"</span><span class="op">]</span>
<span class="va">celltypes</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">markers_scran</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span> 
<span class="va">sign_list</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">celltypes</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
    <span class="va">sign_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="va">top_markers</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">top_markers</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">celltypes</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene</span>
<span class="op">}</span>

<span class="va">PAGE_matrix</span> <span class="op">=</span> <span class="fu"><a href="../../reference/makeSignMatrixPAGE.html">makeSignMatrixPAGE</a></span><span class="op">(</span>sign_names <span class="op">=</span> <span class="va">celltypes</span>,
                                      sign_list <span class="op">=</span> <span class="va">sign_list</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runPAGEEnrich.html">runPAGEEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
                                 sign_matrix <span class="op">=</span> <span class="va">PAGE_matrix</span>,
                                 min_overlap_genes <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="va">cell_types_subset</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">PAGE_matrix</span><span class="op">)</span>
<span class="fu"><a href="../../reference/spatCellPlot.html">spatCellPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, 
             spat_enr_names <span class="op">=</span> <span class="st">'PAGE'</span>,
             cell_annotation_values <span class="op">=</span> <span class="va">cell_types_subset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span>,coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1.25</span>,
             save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"5a_PAGE_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/5a_PAGE_plot.png" style="width:50.0%"></p>
</div>
<div id="hypergeometric-test" class="section level3">
<h3 class="hasAnchor">
<a href="#hypergeometric-test" class="anchor"></a>5.2 Hypergeometric test</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Modify the sparse matrix in normalized slot</span>
<span class="va">testcombo</span><span class="op">@</span><span class="va">expression</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">normalized</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">@</span><span class="va">expression</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">normalized</span><span class="op">)</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runHyperGeometricEnrich.html">runHyperGeometricEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,
                                    expression_values <span class="op">=</span> <span class="st">"normalized"</span>,
                                    sign_matrix <span class="op">=</span> <span class="va">PAGE_matrix</span><span class="op">)</span>
<span class="va">cell_types_subset</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">PAGE_matrix</span><span class="op">)</span>
<span class="fu"><a href="../../reference/spatCellPlot.html">spatCellPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, 
             spat_enr_names <span class="op">=</span> <span class="st">'hypergeometric'</span>,
             cell_annotation_values <span class="op">=</span> <span class="va">cell_types_subset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span>,coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1.75</span>,
             save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"5b_HyperGeometric_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/5b_HyperGeometric_plot.png" style="width:50.0%"></p>
</div>
<div id="rank-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#rank-enrichment" class="anchor"></a>5.3 Rank Enrichment</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create rank matrix, not that rank matrix is different from PAGE</span>
<span class="co"># A count matrix and a vector for all cell labels will be needed</span>
<span class="va">rank_matrix</span> <span class="op">=</span> <span class="fu"><a href="../../reference/makeSignMatrixRank.html">makeSignMatrixRank</a></span><span class="op">(</span>sc_matrix <span class="op">=</span> <span class="va">giotto_SC</span><span class="op">@</span><span class="va">expression</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">normalized</span>,
                                 sc_cluster_ids <span class="op">=</span> <span class="fu"><a href="../../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span><span class="op">$</span><span class="va">prostate_label</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">rank_matrix</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="../../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span><span class="op">$</span><span class="va">prostate_label</span><span class="op">)</span><span class="op">)</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runRankEnrich.html">runRankEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, sign_matrix <span class="op">=</span> <span class="va">rank_matrix</span>,expression_values <span class="op">=</span> <span class="st">"normalized"</span><span class="op">)</span>
<span class="fu"><a href="../../reference/spatCellPlot2D.html">spatCellPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, 
             spat_enr_names <span class="op">=</span> <span class="st">'rank'</span>,
             cell_annotation_values <span class="op">=</span> <span class="va">cell_types_subset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span>,coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1</span>,
             save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"5c_Rank_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/5c_Rank_plot.png" style="width:50.0%"></p>
</div>
<div id="deconvolution" class="section level3">
<h3 class="hasAnchor">
<a href="#deconvolution" class="anchor"></a>5.4deconvolution</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create DWLS matrix, not that DWLS matrix is different from PAGE and rank</span>
<span class="co"># A count matrix a vector for a list of gene signatures and a vector for all cell labels will be needed</span>
<span class="va">DWLS_matrix</span><span class="op">&lt;-</span><span class="fu"><a href="../../reference/makeSignMatrixDWLSfromMatrix.html">makeSignMatrixDWLSfromMatrix</a></span><span class="op">(</span>matrix <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">@</span><span class="va">expression</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">normalized</span><span class="op">)</span>, 
                                cell_type <span class="op">=</span> <span class="fu"><a href="../../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span><span class="op">$</span><span class="va">prostate_label</span>,
                                sign_gene <span class="op">=</span> <span class="va">top_markers</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>
<span class="va">testcombo</span> <span class="op">=</span> <span class="fu"><a href="../../reference/runDWLSDeconv.html">runDWLSDeconv</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, sign_matrix <span class="op">=</span> <span class="va">DWLS_matrix</span><span class="op">)</span>
<span class="fu"><a href="../../reference/spatCellPlot2D.html">spatCellPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, 
             spat_enr_names <span class="op">=</span> <span class="st">'DWLS'</span>,
             cell_annotation_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="../../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span><span class="op">$</span><span class="va">prostate_label</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span>,coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1</span>,
             save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"5d_DWLS_plot"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../inst/images/visium_prostate_integration/vignette_sep29_2021/5d_DWLS_plot.png" style="width:50.0%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.drieslab.com/">Ruben Dries</a>, Jiaji George Chen, Natalie Del Rossi, Emma Kelley, Junxiang Xu, Guo-Cheng Yuan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
